CharlieBackend.Business/Services/AccountService.cs

```diff
@@ -80,17 +80,17 @@ public async Task<AccountDto> GetAccountCredentialsAsync(AuthenticationDto authe
 
                 var foundAccount = _mapper.Map<AccountDto>(await _unitOfWork.AccountRepository.GetAccountCredentials(authenticationModel));
 
-                return foundAccount;
+                return Result<AccountDto>.GetSuccess(foundAccount);
             }
 
-            return null;
+            return Result<AccountDto>.GetError(ErrorCode.NotFound,"User Not Found");
         }
 
-        public async Task<IList<AccountDto>> GetAllAccountsAsync()
+        public async Task<Result<IList<AccountDto>>> GetAllAccountsAsync()
```

Comment:
Don't forget to call .ToActionResult in controller action. Otherwise, it will change schema of response body.